Sources/TuistKit/Services/TestService.swift

```diff
@@ -209,6 +217,11 @@ final class TestService {
             version: version,
             deviceName: deviceName
         )
+        
+        let extraTestArguments: [XcodeBuildArgument] = [
+            retryTestsOnFailure ? .xcflag("-retry-tests-on-failure") : nil,
```

Comment:
I opted to construct these arguments here rather than extend XcodeBuildArgument with them because these arguments only apply to the `TestCommand`. Happy to incorporate them into the enum if you'd prefer.